hazelcast/src/main/java/com/hazelcast/sql/impl/exec/scan/index/IndexEqualsFilter.java

```diff
@@ -49,7 +49,7 @@ public IndexFilterValue getValue() {
     }
 
     @Override
-    public Iterator<QueryableEntry> getEntries(InternalIndex index, ExpressionEvalContext evalContext) {
+    public Iterator<QueryableEntry> getEntries(InternalIndex index, boolean descending, ExpressionEvalContext evalContext) {
```

Comment:
Perhaps the `descending` flag should be propagated to the index iterator, otherwise, we may lose the collation. Do we have tests that ensure that the values returned from different index lookups (equals, in, range) follow the expected collation?

Reply:
The `index.getSqlRecordIterator(value)` returns entries with the same index values so that they're already pre-sorted descending and ascending. This is why we don't propagate the descending `flag` downwards to the index.